File: ejerc.diccionarios/Ejercicio 6/Ejercicio6.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class DiccionarioPersona {
    public static void main(String[] args) {
        // Crear un HashMap para almacenar los datos de la persona
        Map<String, String> persona = new HashMap<>();
        
        // Crear objeto Scanner para leer entradas desde la consola
        Scanner sc = new Scanner(System.in);

        // Bucle infinito para permitir al usuario agregar varios datos
        while (true) {
            // Solicitar el nombre del dato que se quiere agregar
            System.out.print("Ingrese el nombre del dato (o 'salir' para terminar): ");
            String clave = sc.nextLine().trim(); // Elimina espacios extras

            // Si el usuario escribe 'salir' (sin importar mayúsculas/minúsculas), termina el bucle
            if (clave.equalsIgnoreCase("salir")) {
                break;
            }

            // Solicitar el valor asociado a la clave ingresada
            System.out.print("Ingrese el valor para '" + clave + "': ");
            String valor = sc.nextLine().trim();

            // Guardar el par clave-valor en el HashMap
            persona.put(clave, valor);

            // Mostrar el diccionario actualizado
            System.out.println("\nDiccionario actualizado:");
            System.out.println(persona);
        }

        // Cerrar el objeto Scanner para liberar recursos
        sc.close();
    }
}

File: ejerc.diccionarios/Ejercicio 6/Ejercicio6.py
# Crear un diccionario vacío para almacenar los datos de la persona
persona = {}

# Bucle infinito que se repetirá hasta que el usuario escriba 'salir'
while True:
    # Solicitar el nombre de la clave que se desea agregar al diccionario
    clave = input("Ingrese el nombre del dato (o 'salir' para terminar): ").strip()
    
    # Si el usuario escribe 'salir' (en minúsculas, gracias a .lower()), se termina el ciclo
    if clave.lower() == 'salir':
        break

    # Solicitar el valor para la clave ingresada
    valor = input(f"Ingrese el valor para '{clave}': ").strip()
    
    # Agregar la clave y el valor al diccionario
    persona[clave] = valor
    
    # Mostrar el diccionario actualizado después de cada inserción
    print("\nDiccionario actualizado:")
    print(persona)
